impl/src/main/java/org/ehcache/internal/store/tiering/CacheStore.java

```diff
@@ -82,6 +92,11 @@ public void onInvalidation(K key, ValueHolder<V> valueHolder) {
           }
         }
       });
+      if(valueHolder != null && valueHolder.isExpired(timeSource.getTimeMillis(), TimeUnit.MILLISECONDS)) {
+        authoritativeTier.flush(key, valueHolder); // so it gets removed from the authority
+        return null;
```

Comment:
I think that's actually what triggered the refactoring of the third commit, none of that is actually required now anymore _I think_... let me check that. But good catch too...